internal/engine/build_and_deployer.go

```diff
@@ -65,15 +66,33 @@ func (l localBuildAndDeployer) BuildAndDeploy(ctx context.Context, service model
 		return nil, err
 	}
 
+	ref, err := l.build(ctx, service, token, changedFiles)
+	if err != nil {
+		return nil, err
+	}
+
+	k8sEntities, err := l.deploy(ctx, service, ref)
```

Comment:
i'm not sure how I feel about calling this "deploy", imo the actual deploy is applying the config to k8s below?

Reply:
good point! moved below